src/app/date-picker/date-picker.component.spec.ts

```diff
@@ -1,9 +1,64 @@
 import {DatePickerComponent} from './date-picker.component';
+import {DatePickerService} from './date-picker.service';
+import {async, ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {DebugElement, NO_ERRORS_SCHEMA} from '@angular/core';
+import {UtilsService} from '../common/services/utils/utils.service';
+import {DomHelper} from '../common/services/dom-appender/dom-appender.service';
+import {By} from '@angular/platform-browser';
+import {DayCalendarComponent} from '../day-calendar/day-calendar.component';
 
 describe('Component: DpDayPicker', () => {
```

Comment:
this type of test should be in e2e

Reply:
I can't get protractor to trigger `focus` event, only `click` is supported. Also importing dependencies from my `spec` don't compile in e2e tests.

Reply:
OK, lets add the click tests to e2e and I will think about something for the focus case